modules/standard/IO.chpl

```diff
@@ -3010,19 +3010,59 @@ private inline proc _write_binary_internal(_channel_internal:qio_channel_ptr_t,
   return EINVAL;
 }
 
-// Channel must be locked, must be running on this.home
-// x is ref (vs out) because it might contain a literal string.
+pragma "no doc"
+proc channel._constructIoErrorMsg(param kind: iokind, const x:?t): string {
+  var result: string = "while ";
+
+  result += if this.writing then "writing " else "reading ";
+  result += t:string;
+
+  select t {
+    when ioNewline do result += " " + "newline";
+    when ioLiteral do result += " " + "\"" + x:string + "\""; 
+  }
+
+  return result;
+}
+
+//
+// The channel must be locked and running on this.home.
+// The intent of x is ref (vs out) because it might contain a string literal.
+//
+pragma "no doc"
+proc channel._readOne(param kind: iokind, ref x:?t,
```

Comment:
Let's put a comment like "TODO: incorporate this logic into _read_one_internal and make it a method"